### PR TITLE
tests(span-summary): Add test coverage for new Span Summary page

### PR DESCRIPTION
**Pull request by 0Calories**
Adds some initial basic tests to the Span Summary page. Includes some slight refactors to allow for testing via hooks and adding `data-test-id`'s

For the sake of reviewability, I will be adding more tests in followup PRs

### CODE REVIEW

File: static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        project={undefined}
+      />
+    );
+
+    expect(headerDataMock).toHaveBeenCalled();
+
+    const headerContainerOp = await screen.findByTestId('operation-name');
+    const headerContainerDescription = await screen.findByTestId(
+      'header-span-description'
+    );
```
Comment (gggritso) [inline static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx:269]: If this is a header, might be better to do something like `screen.getByRole('header', {level: 2})` and check its contents (sorry, I forgot the exact syntax, but the idea is that querying by role is more flexible)

File: static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    const timeSpent = await screen.findByTestId('header-total-time-spent');
+    const totalSpanCount = await screen.findByTestId('total-span-count');
+
+    expect(headerContainerOp).toHaveTextContent('db');
+    expect(headerContainerDescription).toHaveTextContent(
+      'SELECT thing FROM my_cool_db WHERE value = %s'
+    );
+    expect(avgDuration).toHaveTextContent('1.74ms');
+    expect(timeSpent).toHaveTextContent('2.43mo');
+    expect(totalSpanCount).toHaveTextContent('3.6b spans');
```
Comment (gggritso) [inline static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx:280]: Totally fine to do this on one line, e.g.,

```
expect(await screen.findByTestId('total-span-count')).toHaveTextContent(whatever)
```

which is a bit more dense

**Review by gggritso — APPROVED** [review]
Cool! A few nits, but more test coverage is great 👍🏻

File: static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+import {useParams} from 'sentry/utils/useParams';
+import useProjects from 'sentry/utils/useProjects';
+import SpanSummary from 'sentry/views/performance/transactionSummary/transactionSpans/spanSummary/content';
+
+jest.mock('sentry/utils/useParams');
+jest.mock('sentry/utils/useLocation');
+jest.mock('sentry/utils/usePageFilters');
+jest.mock('sentry/utils/useProjects');
+
+describe('Performance > Transaction Spans > Span Summary', function () {
```
[the remark this answers is not in the document]
Comment (0Calories) [reply static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx:17]: Nice, that's good to know! I copied the test name from the previous test file, but if the test runner automatically sets the hierarchy for us then it makes a lot more sense to name it differently. I've changed the name of this test and will rename the component itself in a followup PR.

File: static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+    render(
+      <SpanSummary
+        spanSlug={{group: 'aaaaaaaa', op: 'db'}}
+        transactionName="transaction"
+        organization={organization}
+        project={undefined}
+      />
+    );
+
+    expect(headerDataMock).toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (0Calories) [reply static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx:264]: Oh great catch, I had some before but removed them when I was debugging some funky business and forgot to put it back. At the bottom of the test you can actually see the expect call that is currently in a useless spot 😂

File: static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx (hunk 1 above)
Comment (0Calories) [reply static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx:269]: It's built with just divs so I can't use this unfortunately, but I'll keep that in mind when writing future tests!

**Comment by 0Calories** [comment]
Great suggestions, thanks @gggritso !

**Pull request merged**